lms/djangoapps/student_account/views.py

```diff
@@ -302,6 +311,45 @@ def _external_auth_intercept(request, mode):
         return external_auth_register(request)
 
 
+def _get_order_details(user):
+    """ Retrieve order details for the user. """
+    cache_key = CommerceConfiguration.CACHE_KEY + '.' + user.username
+    cached = cache.get(cache_key)
+    if cached:
+        return cached
+
+    order_details = []
+    allowed_certificates = ['professional', 'verified', 'credit']
+    try:
+        commerce_order_details = ecommerce_api_client(user).orders.get()
+    except Exception:  # pylint: disable=broad-except
+        log.exception('Failed to retrieve data from the Commerce API.')
+        return None
+
+    for order in commerce_order_details['results']:
+        if order['status'].lower() == 'complete':
+            for attribute in order['lines'][0]['product']['attribute_values']:
+                if attribute['name'] == 'certificate_type' and attribute['value'] in allowed_certificates:
+                    try:
+                        order_data = {
+                            'number': order['number'],
+                            'price': order['total_excl_tax'],
+                            'title': order['lines'][0]['title'],
+                            'order_date': ModuleI18nService().strftime(
+                                datetime.strptime(order['date_placed'], "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=pytz.UTC),
+                                'SHORT_DATE'
+                            ),
+                            'receipt_url': '/commerce/checkout/receipt/?orderNum=' + order['number']
+                        }
+                        order_details.append(order_data)
+                    except KeyError:
+                        log.warning('order structure is invalid: %r', order)
+                        return None
+
+    cache.set(cache_key, order_details, CommerceConfiguration.CACHE_TTL)
```

Comment:
before setting the cache first check if the cache is enabled (i.e. the value is greater than 0) or not.